### PR TITLE
A32/T32/A64: Implement Hint instructions (CSDB, SEV, SEVL, WFE, WFI, YIELD)

### LINKED ISSUE #3686: Unimplemented SEV instruction
### What CPU instruction is missing?

```
The instruction at 0x00000000003357f4 (opcode 0xe320f004) is undefined!
```

### Instruction name
```
0x0000000000000000:  04 F0 20 E3    sev 
```

### Required by: 
Hanayaka Nari Waga Ichizoku Modern Nostalgie [01008DE00C022000] [32-bit] (no game issue exists)
Original issue here: https://github.com/Ryujinx/Ryujinx/issues/2717

### PR DESCRIPTION
**Pull request by merryhime**
Implement as nops.

Closes #3686.

### CODE REVIEW

**Comment by gdkchan** [comment]
You should probably remove the existing A64 catch-all hint instruction: https://github.com/Ryujinx/Ryujinx/blob/master/ARMeilleure/Decoders/OpCodeTable.cs#L110

**Comment by merryhime** [comment]
Would you prefer the other way around? (adding a catch all to all instruction sets?)

**Comment by gdkchan** [comment]
> Would you prefer the other way around? (adding a catch all to all instruction sets?)

Personally I prefer having them separate.
And eventually we should probably implement WFE/SEV/SEVL since otherwise the CPU will be spin waiting (which is technically still correct, but not optimal).

**Comment by merryhime** [comment]
> And eventually we should probably implement WFE/SEV/SEVL since otherwise the CPU will be spin waiting (which is technically still correct, but not optimal).

My fear is introducing unintentional deadlocks due to beibng overenthusiastic here. I think a simple thing to do is to do a host OS thread yield on WFE/WFI when required (since afaik Ryujinx is 1:1 with host:emu threads).

File: ARMeilleure/Decoders/OpCodeTable.cs [hunk 1]
```diff
@@ -107,7 +108,6 @@ static OpCodeTable()
             SetA64("11001010xx0xxxxxxxxxxxxxxxxxxxxx", InstName.Eor,             InstEmit.Eor,             OpCodeAluRs.Create);
             SetA64("00010011100xxxxx0xxxxxxxxxxxxxxx", InstName.Extr,            InstEmit.Extr,            OpCodeAluRs.Create);
             SetA64("10010011110xxxxxxxxxxxxxxxxxxxxx", InstName.Extr,            InstEmit.Extr,            OpCodeAluRs.Create);
-            SetA64("11010101000000110010xxxxxxx11111", InstName.Hint,            InstEmit.Hint,            OpCodeSystem.Create);
```
Comment (gdkchan) [inline ARMeilleure/Decoders/OpCodeTable.cs:110]: The enum entry and the Hint function can be removed too.

**Review by gdkchan — APPROVED** [review]
lgtm, thanks!

File: ARMeilleure/Decoders/OpCodeTable.cs [hunk 2]
```diff
@@ -1060,7 +1070,14 @@ static OpCodeTable()
...
             SetT16("101110x1xxxxxxxx", InstName.Cbnz,   InstEmit32.Cbnz,    OpCodeT16BImmCmp.Create);
             SetT16("1011110xxxxxxxxx", InstName.Pop,    InstEmit32.Ldm,     OpCodeT16MemStack.Create);
-            SetT16("10111111xxxx0000", InstName.Nop,    InstEmit32.Nop,     OpCodeT16.Create);
+            SetT16("1011111100000000", InstName.Nop,    InstEmit32.Nop,     OpCodeT16.Create);
+            SetT16("1011111100010000", InstName.Yield,  InstEmit32.Nop,     OpCodeT16.Create);
+            SetT16("1011111100100000", InstName.Wfe,    InstEmit32.Nop,     OpCodeT16.Create);
+            SetT16("1011111100110000", InstName.Wfi,    InstEmit32.Nop,     OpCodeT16.Create);
+            SetT16("1011111101000000", InstName.Sev,    InstEmit32.Nop,     OpCodeT16.Create);
+            SetT16("1011111101010000", InstName.Sevl,   InstEmit32.Nop,     OpCodeT16.Create);
+            SetT16("10111111011x0000", InstName.Hint,   InstEmit32.Nop,     OpCodeT16.Create); // Hint instruction
```
Comment (gdkchan) [inline ARMeilleure/Decoders/OpCodeTable.cs:1079]: Are there no unallocated hint instructions on A32 and A64?

Comment (merryhime) [reply ARMeilleure/Decoders/OpCodeTable.cs:1079]: They do exist, I could do the same.

**Review by Thog — APPROVED** [review]

**Comment by gdkchan** [comment]
Needs a rebase.

**Comment by merryhime** [comment]
rebased

**Pull request merged**